guiEditor/src/diagram/workbench.tsx

```diff
@@ -80,19 +83,50 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
             }
         });
 
-        props.globalState.onPanObservable.add( () => {
-            this._forcePanning = true;
+        props.globalState.onPanObservable.add(() => {
+            this._forcePanning = ! this._forcePanning;

```

Comment:
same thing with the component state - every time you need a class variable that will be used in the template, think "state".